project-base/tests/ShopBundle/Database/EntityExtension/EntityExtensionTest.php

```diff
@@ -487,4 +495,88 @@ private function getOrderItem(int $id): ExtendedOrderItem
         $this->assertInstanceOf(ExtendedOrderItem::class, $result);
         return $result;
     }
+
+    private function doTestExtendedProductInstantiation(): void
+    {
+        $repository = $this->em->getRepository(Product::class);
+        $this->assertInstanceOf(ExtendedProduct::class, $repository->find(self::MAIN_PRODUCT_ID));
+
+        $query = $this->em->createQuery('SELECT p FROM ' . Product::class . ' p WHERE p.id = :id')
+            ->setParameter('id', self::MAIN_PRODUCT_ID);
+        $this->assertInstanceOf(ExtendedProduct::class, $query->getSingleResult());
+
+        $qb = $this->em->createQueryBuilder();
+        $qb->from(Product::class, 'p')
+            ->select('p')
+            ->where('p.id = :id')
+            ->setParameter(':id', self::MAIN_PRODUCT_ID);
+        $this->assertInstanceOf(ExtendedProduct::class, $qb->getQuery()->getSingleResult());
+
+        $product = $this->em->find(Product::class, self::MAIN_PRODUCT_ID);
+        $this->assertInstanceOf(ExtendedProduct::class, $product);
+    }
+
+    private function doTestExtendedCategoryInstantiation(): void
+    {
+        $repository = $this->em->getRepository(Category::class);
+        $this->assertInstanceOf(ExtendedCategory::class, $repository->find(self::MAIN_CATEGORY_ID));
+
+        $query = $this->em->createQuery('SELECT c FROM ' . Category::class . ' c WHERE c.id = :id')
+            ->setParameter('id', self::MAIN_CATEGORY_ID);
+        $this->assertInstanceOf(ExtendedCategory::class, $query->getSingleResult());
+
+        $qb = $this->em->createQueryBuilder();
+        $qb->from(Category::class, 'c')
+            ->select('c')
+            ->where('c.id = :id')
+            ->setParameter(':id', self::MAIN_CATEGORY_ID);
+        $this->assertInstanceOf(ExtendedCategory::class, $qb->getQuery()->getSingleResult());
+
+        $category = $this->em->find(Category::class, self::MAIN_CATEGORY_ID);
+        $this->assertInstanceOf(ExtendedCategory::class, $category);
+    }
+
+    private function doTestExtendedOrderItemsInstantiation(): void
```

Comment:
I would split this method into three smaller in order to increase readability of the code.

Reply:
That might help. @vitek-rostislav, could you split it, please?

Reply:
:+1: added fixup commit :slightly_smiling_face: